--- a/techpack/display/msm/sde/sde_hw_dsc_1_2.c
+++ b/techpack/display/msm/sde/sde_hw_dsc_1_2.c
@@ -77,7 +77,7 @@ static int _dsc_calc_ob_max_addr(struct sde_hw_dsc *hw_dsc, int num_ss)
 	return 0;
 }
 
-static inline _dsc_subblk_offset(struct sde_hw_dsc *hw_dsc, int s_id,
+static inline int _dsc_subblk_offset(struct sde_hw_dsc *hw_dsc, int s_id,
 		u32 *idx)
 {
 	const struct sde_dsc_sub_blks *sblk;
