diff --git a/lib/Makefile b/lib/Makefile
index 685aee60de89..81be0b2c47c9 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -223,6 +223,7 @@ obj-$(CONFIG_IRQ_POLL) += irq_poll.o
 
 libfdt_files = fdt.o fdt_ro.o fdt_wip.o fdt_rw.o fdt_sw.o fdt_strerror.o \
 	       fdt_empty_tree.o
+libfdt_files += $(if $(CONFIG_KEXEC_FILE),fdt_addresses.o)
 $(foreach file, $(libfdt_files), \
 	$(eval CFLAGS_$(file) = -I $(srctree)/scripts/dtc/libfdt))
 lib-$(CONFIG_LIBFDT) += $(libfdt_files)
